client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/network/TransportFrameDecoderWithBufferSupplier.java

```diff
@@ -123,8 +124,15 @@ private io.netty.buffer.ByteBuf decodeBodyCopyOut(
       return buf;
     }
 
-    ReadData readData = (ReadData) curMsg;
-    long streamId = readData.getStreamId();
+    long streamId;
+    if (curMsg instanceof ReadData) {
```

Comment:
Could `SubPartitionReadData` extend `ReadData`?

Reply:
> Could `SubPartitionReadData` extend `ReadData`?

Okay, I will try to make SubPartitionReadData extend ReadData.

Reply:
Hi, @SteNicholas

I am not sure if i can modify `org.apache.celeborn.plugin.flink.protocol.ReadData` to a non-final class?